lexnlp/extract/batch/tests/test_batch_init.py

```
@@ -92,4 +92,109 @@ async def _run():
+        import lexnlp.extract.batch as batch_pkg
+
+        assert "extract_batch_with_progress" not in batch_pkg.__all__
+
+    def test_adaptive_max_workers_not_importable_from_batch(self) -> None:
+        """
+        Verifies that importing `adaptive_max_workers` from `lexnlp.extract.batch` raises ImportError.
+        
+        This test ensures `adaptive_max_workers` is not exposed as a public import from the `lexnlp.extract.batch` package.
+        """
+        with pytest.raises(ImportError):
+            from lexnlp.extract.batch import adaptive_max_workers  # noqa: F401
```

Pylint (4.0.5)

[error] No name 'adaptive_max_workers' in module 'lexnlp.extract.batch'

(E0611)